Fix Socket reading of abstract unix domain addresses

normanmaurer opened the pull request:
Motivation:

How we calculated the length of the unix domain socket address was not correct.

Motification:

Follow the manpage in how the length is calculated

Result:

Always return the correct name for abstract unix domain sockets

normanmaurer committed:
Fix Socket reading of abstract unix domain addresses

Motivation:

How we calculated the length of the unix domain socket address was not correct.

Motification:

Follow the manpage in how the length is calculated

Result:

Always return the correct name for abstract unix domain sockets

normanmaurer committed:
Extract to helper function to reduce duplication

normanmaurer committed:
Add more asserts

chrisvest commented on transport-native-unix-common/src/main/c/netty_unix_socket.c:
```diff
@@ -133,23 +133,24 @@ static jobject createDatagramSocketAddress(JNIEnv* env, const struct sockaddr_st
 #else
-    int pathLength = strlen(s->sun_path);
+    return strlen(s->sun_path);
 #endif
```
```suggestion
#endif
    return strlen(s->sun_path);
```

chrisvest replied:
The diff above looks kind of weird. Basically, having `#else` is wrong. We should always unconditionally fall back to `strlen(s->sun_path)`

chrisvest reviewed (changes requested):
Spotted a bug

chrisvest committed:
Avoid having a codepath where we don't return any domain socket path length.

chrisvest committed:
Avoid negative array size on abstract domain path

Abstract domain socket paths are always at least sized
sizeof(sa_family_t) or greater.
